### ISSUE TITLE
false positive "assigning-non-slot" error for inherited descriptor when slots are used

### ISSUE THREAD

**Issue by jab** [PyCQA/pylint#6001]
### Bug description

Minimized reproducer:
```python
"""Repro false positive bug in pylint."""


# pylint: disable=too-few-public-methods


class MyDescriptor:
    """Basic descriptor."""

    def __get__(self, instance, owner):
        return 42

    def __set__(self, instance, value):
        pass


class Base:
    """Base class with slots."""

    __slots__ = ()

    attr2 = MyDescriptor()


class Repro(Base):
    """Reproduce false positive error using this class."""

    __slots__ = ()


repro = Repro()
repro.attr2 = 'false positive "assigning-non-slot" error'  # false positive error here
```

A real-world (non-minimized) example where pylint produces this false positive error can be found on this line: https://github.com/jab/bidict/blob/caf703e959ed4471bc391a7794411864c1d6ab9d/bidict/_orderedbase.py#L101

### Command used

```shell
pylint repro.py
```

### Pylint output

```shell
************* Module repro
repro.py:32:0: E0237: Assigning to attribute 'attr2' not defined in class slots (assigning-non-slot)
```

### Expected behavior

No error should be flagged here.

### Pylint version

```shell
pylint 2.13.2
astroid 2.11.2
Python 3.10.2 (main, Feb  2 2022, 07:36:01) [Clang 12.0.0 (clang-1200.0.32.29)]
```

**Comment by mbyrnepr2** [comment]
Regarding the provided example pasted above, I think this is expected Pylint behaviour due to `attr2` not being one of the slots.  

`__slots__ = ("attr2",)` vs. `__slots__ = ()`

**Comment by mbyrnepr2** [comment]
The answer [here](https://stackoverflow.com/questions/4912499/using-python-descriptors-with-slots) and [here](https://stackoverflow.com/questions/65537008/how-to-use-descriptors-and-slots-in-python-at-the-same-time) may be useful to achieve what you are looking for.

**Comment by jab** [comment]
Slots are for instance attributes. In this example, `attr2` is (and should be) a class attribute, therefore it is omitted from slots, as it should be. Pylint is incorrectly treating it as an instance attribute, not the inherited class attribute that it actually is.

**Comment by mbyrnepr2** [comment]
Thanks for the info @jab. I certainly don't disagree with your analysis but the behaviour is also seen with the simpler case (irrespective of class attribute descriptors):
```python
class Example:
    __slots__ = ('a',)


myinstance = Example()

myinstance.b = 1 # a.py:7:0: E0237: Assigning to attribute 'b' not defined in class slots (assigning-non-slot)
```

Perhaps a maintainer can clarify what the intention is from the checker in this situation.

**Comment by jab** [comment]
@mbyrnepr2, your simplified example is not the same as what's going on here. In your example, you are assigning an instance attribute `b`. Since `b` is not present in slots for `Example` (or any of its base classes), pylint is correctly flagging the `assigning-non-slot` error.

This issue is about pylint incorrectly flagging an `assigning-non-slot` error when an inherited descriptor is used. If you try to simplify the example I gave in the description to a case where a non-inherited descriptor is used, you can see that pylint correctly does not flag any `assigning-non-slot` error:
```python
class MyDescriptor:
    def __get__(self, instance, owner):
        return 42
    def __set__(self, instance, value):
        instance.a = 42


class Example:
    __slots__ = ('a',)

    b = MyDescriptor()


myinstance = Example()
myinstance.b = 'set a to 42'  # no error on this line, as we expect
print(myinstance.a)  # prints 42
```

**Comment by mbyrnepr2** [comment]
@jab I agree with you. It's an example of correct usage vs. what you are reporting. Sorry that I didn't communicate that clearly.

**Comment by Hubrrr88** [comment]
I have seen the same issue on latest PyLint version 2.13.8. Any news on when will this be fixed yet?

**Comment by clavedeluna** [comment]
It appears this case was considered and tested [here](https://github.com/PyCQA/pylint/blob/main/tests/functional/a/assigning/assigning_non_slot.py#L132) and deemed to need to emit this message. But I believe this is incorrect, as __slots__ is meant for instance attributes, not class attributes.
